### ISSUE TITLE
Incorrect time zone conversion when date.toLocaleString() ends in "PM"

### ISSUE THREAD

**Issue by AtriaGwanfried** [Hexagon/minitz#19]
**Environment:** croner 6.0.2 (latest release), minitz 4.0.0
node v12.9.0, ts-node-dev ver. 1.1.8 (using ts-node ver. 9.1.1, typescript ver. 3.9.10)
Local time zone: `America/New_York`

**Context:** Found this bug while debugging a `croner` start time not being at the expected time, in a provided time zone:

```
const subscriptionJob = croner("0 5 0 * * *",
  {
    timezone: "America/Los_Angeles"
  }, async function () {
    console.log("Function body!");
  })
```

The cited `subscriptionJob.nextRun()` for this was not `Thu Mar 09 2023 3:05:00 GMT-0500 (Eastern Standard Time)` (12:05 AM PST), but instead `Thu Mar 09 2023 12:05:00 GMT-0500 (Eastern Standard Time)`. Eventually I traced the issue to likely be in this library.

**Attempted Identification of Cause**

The potential cause was identified through Visual Studio code debugging, attaching a debugger to the running process shortly before the `croner` construction, and stepping through the `croner` and `minitz` code. Attempting to reproduce this with a unit test failed (see last section).

When a `minitz` date is constructed with a string timezone (e.g. `minitz(2023, 2, 9, 0, 5, 0, "America/Los_Angeles", undefined)`), the library begins by constructing an equivalent JavaScript Date object in the current local timezone (for this example, "America/New_York"):

https://github.com/Hexagon/minitz/blob/aa33229eb0170b9b9e6516fac8fdcdb05448aae5/src/minitz.js#L99

`inDate.getTime() === 1678320300000`(Wed Mar 08 2023 19:05:00 GMT-0500 (Eastern Standard Time) ) , as expected.

The library then gets the offset between the local timezone and UTC:

https://github.com/Hexagon/minitz/blob/aa33229eb0170b9b9e6516fac8fdcdb05448aae5/src/minitz.js#L109

A bug occurs here where because `date.toLocaleString("en",{timeZone:"America/Los_Angeles",timeStyle:"long"})` does not have a timezone, e.g. `"3/8/2023, 4:05:00 PM"`, **the `.slice(-1)[0]` returns "PM" instead of "PST" resulting in `tz = "PM"` and `Date.parse("${dateString} ${tz}")` parsing a string like `3/8/2023, 7:05:00 PM PM`**. This does not cause a problem immediately, as it produces an equivalent result to `3/8/2023, 7:05:00 PM`.

https://github.com/Hexagon/minitz/blob/aa33229eb0170b9b9e6516fac8fdcdb05448aae5/src/minitz.js#L109

`offset = -18000000` (-5 hours), as expected.
`dateGuess.getTime() === 1678338300000` (Thu Mar 09 2023 00:05:00 GMT-0500 (Eastern Standard Time) )

The actual bug occurs in the calculation of the offset of the date guess object:

https://github.com/Hexagon/minitz/blob/aa33229eb0170b9b9e6516fac8fdcdb05448aae5/src/minitz.js#L115

`return Date.parse("3/9/2023, 12:05:00 AM GMT") - Date.parse("3/9/2023, 12:05:00 AM PM")`

`dateOffsGuess = -61200000` (-17 hours)

Now the extra "PM" is a problem, where the extra PM effectively converts the time 12:05 AM to 12:05 PM. This causes the "second try" logic to produce a timestamp that is misaligned by 12 hours.

**Attempted Reproduction**

Adding a new unit test, and running it in local timezone `America/New_York` somehow produces expected results:

```
	test("Timezone conversion crossing a day boundary", function () {
		const justAfterMidnightPST_UTCForm = new Date(Date.UTC(2023,2,9,8,5,0));
		const justAfterMidnightPST_minitz = minitz(2023, 3, 9, 0, 5, 0, "America/Los_Angeles");

		// new Date(Date.parse("3/9/2023, 12:05:00 AM PST")).getTime() === 1678349100000
		assert.equal(justAfterMidnightPST_UTCForm.getTime(), justAfterMidnightPST_minitz.getTime());
		assert.equal(justAfterMidnightPST_UTCForm.getTime(), 1678349100000);
	});
```

There may thus be something sneaky going on with locales, or debugger attachment - I am not yet sure at this point.

Let me know if I can provide any further info. Thanks for your time.

**Comment by Hexagon** [comment]
Thanks for a great bug report! Will have a look at this asap 👍

**Comment by Hexagon** [comment]
I cannot figure out how to get a date without timezone at the end

![image](https://user-images.githubusercontent.com/419737/223870451-77442e49-3380-4f84-b422-e93bc1433307.png)

Do you have any ideas what could be causing this?

**Comment by Hexagon** [comment]
![image](https://user-images.githubusercontent.com/419737/223879670-427df9a3-892d-4ba7-aad9-cc9944c556dd.png)

**Comment by Hexagon** [comment]
Could ts-node be doing something wierd? Have you been able to reproduce it using "pure node"?

**Comment by Hexagon** [comment]
Oh, i think i've got it. Running

```new Date().toLocaleString("en",{timeZone:"America/Los_Angeles",timeStyle:"long"})```

does not work in ts-node REPL, it doesn't seem to support the timeStyle option. Replacing the line with 

```new Date().toLocaleString("en",{timeZone:"America/Los_Angeles",timeZoneName:"short"})```

works...

Will release a new version of minitz and croner with these changes shortly.

**Comment by Hexagon** [comment]
Proposed solution implemented in `croner@6.0.3-dev.0`

**Comment by AtriaGwanfried** [comment]
Hi there,

Wow, thanks for the incredibly quick turn-around for this, and the good detective work! I re-tested with `croner@6.0.3-dev.0` and confirmed that this does fix the problem I was seeing; the `subscriptionJob.nextRun()` is now reporting `3:05:00 GMT-0500 (Eastern Standard Time)` as expected.

I should be all set now.

**Comment by Hexagon** [comment]
Excellent 👍  This is released in minitz `4.0.5` and will be included in croner `6.0.3`, when it is time.

**Comment by Hexagon** [comment]
Released in croner v`6.0.3`

**Comment by AtriaGwanfried** [comment]
Follow-up update: Can confirm the `6.0.3-dev.0` fix has been working great for the past few weeks. 👍 

We'll upgrade to 6.0.3 stable the next time we do a general package version update for libraries. Thanks again.
